### DISCUSSION TITLE
Installation Support

### DISCUSSION THREAD

**Discussion by ikeyurp** [coder/code-server#3195] [Q&A] [answered]
Hello,
I tried installing the code-server on my ubuntu server. My server already had tomcat installed on port 8080 so when I installed code-server with install script, I am getting an error and it is not starting. I think its port is getting clashed. Any help would be appreciated. I have attached a screenshot for the same.
![code-server](https://user-images.githubusercontent.com/26923503/115821713-eacacc80-a420-11eb-9768-3e589819c7df.png)

**Comment by mrkaushal** [comment]
Same Issue, please help me to figure out

**Comment by nooobcoder** [comment, marked as answer]
Dear @ikeyurp,

First, open this file with any editor like nano or vim

`~/.config/code-server/config.yaml`
_Then change the port number with your desired port._
After opening the file, you will see similar output
![image](https://user-images.githubusercontent.com/50350828/115856312-f2529b80-a449-11eb-8076-e4a0daf35739.png)
Change the port number after the ip address in the `ip-addr` section.

Then, restart the code-server service
`sudo systemctl restart code-server`
`sudo systemctl status code-server`

If everything goes right, you would be able to access code-server at the new port followed by the ip-address. Also, the success logs would be similar.
![image](https://user-images.githubusercontent.com/50350828/115856517-3cd41800-a44a-11eb-87fe-2d9b6add7cb6.png)

@mrkaushal please refer to this answer and you should be good to go.

> @ikeyurp  i am impressed, you are using Termius to connect to the server via SSH, go on, Termius is great for managing ssh connections. I personally use it. Also the sync feature is just great to have the config across different machines.

**Also, if you feel I have answered your question and your query has been resolved, then please mark the discussion as marked, this would help the moderators to mark discussions that do not required to be discussed farther.**

Thank You,
from **Ankur Paul** 🇮🇳

**Reply by ikeyurp** [reply]
I already tried it and facing an address already in use error.

![ss1](https://user-images.githubusercontent.com/26923503/115866375-34360e80-a457-11eb-94b4-093dd6dfff5f.png)
![ss2](https://user-images.githubusercontent.com/26923503/115866365-326c4b00-a457-11eb-8bf5-a4fb19f460df.png)

**Reply by nooobcoder** [reply]
@ikeyurp did you restart the code-server service?
I have mentioned how to do that

**Reply by ikeyurp** [reply]
I tried but getting this error.
![image](https://user-images.githubusercontent.com/26923503/115867358-a2c79c00-a458-11eb-8127-ae6bf808c0df.png)

**Reply by nooobcoder** [reply]
This seems like the service is not available to the current user
Try running 
`sudo systemctl enable --now code-server@$USER`
and then restart the service with 
`sudo systemctl restart code-server`

This might happen if you installed code-server **with a user** and accessing the **code-server with a different user**.

If still does not work, switch to root user, 
by `sudo su`
`sudo systemctl restart code-server`

**Reply by ikeyurp** [reply]
Tried it but the issue persists. I have only 1 user account.
![image](https://user-images.githubusercontent.com/26923503/115868482-23d36300-a45a-11eb-82e6-2597d9f7d7de.png)

**Reply by ikeyurp** [reply]
@nooobcoder Can you guide me through the installation process? I used an installation script for installation. Is there any alternate way to install it and change the port?

**Reply by nooobcoder** [reply]
uninstall code-server first with `sudo apt-get purge code-server`
Then
```
curl -fOL https://github.com/cdr/code-server/releases/download/v3.9.3/code-server_3.9.3_amd64.deb
sudo dpkg -i code-server_3.9.3_amd64.deb
sudo systemctl enable --now code-server@$USER
```

This would download the `deb` fiile from latest build and install it then it bind the service to the current user account.
Can you tell me if this worked? Also, are you using Ubuntu 18 or any other distro?

### FOR FEDORA
```
curl -fOL https://github.com/cdr/code-server/releases/download/v3.9.3/code-server-3.9.3-amd64.rpm
sudo rpm -i code-server-3.9.3-amd64.rpm
sudo systemctl enable --now code-server@$USER
```

### FOR ARCH LINUX
```
yay -S code-server
sudo systemctl enable --now code-server@$USER
```


You can use code-server as a docker container too but that might add complications to you. Let's see with these steps first if not done will try with docker.
Also docker can be an overhead for you to manage the containers.

**Reply by ikeyurp** [reply]
I tried purging code-server and then installing it as you said. I followed the exact steps. Still, service could not be found. Can we explicitly create a service file here? Yes, I am using Ubuntu 20.04
![image](https://user-images.githubusercontent.com/26923503/115869878-220a9f00-a45c-11eb-93d1-b550dd3381d1.png)

**Reply by nooobcoder** [reply]
After this command you need to run `"curl -fsSL https://code-server.dev/install.sh | sh"`
If nothing works, there might be an issue!

Please report back if it worked.

**Reply by ikeyurp** [reply]
I tied the following code and then used ``` systemctl --user enable code-server ``` and it worked. Thank you @nooobcoder for help.

> Uninstall code-server first with `sudo apt-get purge code-server`
> Then
> 
> ```
> curl -fOL https://github.com/cdr/code-server/releases/download/v3.9.3/code-server_3.9.3_amd64.deb
> sudo dpkg -i code-server_3.9.3_amd64.deb
> sudo systemctl enable --now code-server@$USER
> ```

**Reply by nooobcoder** [reply]
Yes @ikeyurp  you were **not enabling the service to the system daemon** with the current user and thus the service was not registered. 
_If you followed each line carefully, the problem would be solved early._

Enjoy using `code-server`, if you have any problem in the future, feel free to ask out in the discussions panel.

Anyways, thank you
from **Ankur Paul** 🇮🇳

**Reply by nooobcoder** [reply]
Also, @mrkaushal did this discussion work out for you?

**Comment by mrkaushal** [comment]
@nooobcoder I am getting the following error while following the above steps.
![image](https://user-images.githubusercontent.com/26923503/115901364-b38b0880-a47e-11eb-983f-c558b3f08a45.png)

**Reply by nooobcoder** [reply]
@mrkaushal please attach a screenshot or the direct text as obtained from the **config.yaml** file.
Use the following command to get the contents of the file
`cat ~/.config/code-server/config.yaml`

**Reply by mrkaushal** [reply]
![image](https://user-images.githubusercontent.com/55987371/115946087-80776200-a4dc-11eb-9196-fca93ea77eb9.png)

**Reply by nooobcoder** [reply]
@mrkaushal can you please try by replacing the **bind-addr** to either `0.0.0.0` or `127.0.0.1`.
After doing so, restart the service by
`sudo systemctl restart code-server`

**Reply by mrkaushal** [reply]
![image](https://user-images.githubusercontent.com/26923503/115946410-88380600-a4de-11eb-9fd1-20c7182c6916.png)

**Reply by nooobcoder** [reply]
@mrkaushal i have explained the steps how to uninstall the code-server and start fresh for debian/arch systems. Please see the above threads as I have already solved this problem for @ikeyurp . If still not successful, reply. He was having the same issue of teh service not being found.

**Reply by mrkaushal** [reply]
I tried it many times but failed. 
![image](https://user-images.githubusercontent.com/26923503/115946597-f7622a00-a4df-11eb-85a3-35acd691501d.png)

**Reply by nooobcoder** [reply]
@mrkaushal  be careful with your commands you are typing the service name incorrectly, it is `code-server`
`sudo systemctl status code-server`

**Reply by mrkaushal** [reply]
Sorry, My bad. 
![image](https://user-images.githubusercontent.com/26923503/115946689-9129d700-a4e0-11eb-8944-66972a8fc5c8.png)

**Reply by nooobcoder** [reply]
@ikeyurp please guide @mrkaushal his problem is similar to you. Guide him how you fixed your issue.
